pfrl/experiments/train_hrl_agent.py

```diff
@@ -29,9 +29,11 @@ def train_hrl_agent(
 
     fg = obs_dict['desired_goal']
     obs = obs_dict['observation']
-
     # sample from subgoal
-    sg = env.subgoal_space.sample()
+    if agent.subgoal_space is not None:
+        sg = agent.subgoal_space.sample()
+    else:
+        sg = env.subgoal_space.sample()
```

Comment:
Aren't they the same?

Reply:
in the fetch env's case, they don't have a built in `subgoal_space` like the AntEnv. So, if that's the case, I give the agent the `subgoal_space`, and then sample from there.